design/k8s-job-hook.md

```diff
@@ -0,0 +1,176 @@
+# Kubernetes Job Hook Proposal
+
+## Goals
+
+- Enable hooks that trigger execution of some behavior directly by the lakeFS server using Kubernetes
+- Enable non-blocking execution for post-merge and post-merge
+- Support successful job completion as condition for the commit and merge operations
+
+
+## Non Goals
+
+- GitHub Actions - fully maintained, virtualized environment with pre-build images to execute pre/post hooks
+
+
+## Proposition
+
+### Overview
+
+New lakeFS hook type, triggered by pre/post commit/merge events to execute jobs on k8s cluster.
```

Comment:
Please link to our docs page describing Actions & Hooks.

Reply:
It should clear that understanding what we have today is a prerequisite to reading this documentation.